### PR TITLE
Add specific JDK 11, 15, and 16 for OldObjectSampleEvent

### PR DESCRIPTION
**Pull request by luhenry**

### CODE REVIEW

**Comment by luhenry** [comment]
@jbachorik I based the JDK versions on https://bugs.openjdk.java.net/browse/JDK-8258414, could you please check they are correct?

File: internal-api/src/main/java/datadog/trace/api/Platform.java [hunk 1]
```diff
@@ -4,33 +4,98 @@
...
+    return new Version(major, minor, update);
+  }
+
+  private static class Version {
+    public final int major, minor, update;
+
+    public Version(int major, int minor, int update) {
+      this.major = major;
+      this.minor = minor;
+      this.update = update;
```
Comment (tylerbenson) [inline internal-api/src/main/java/datadog/trace/api/Platform.java:67]: Isn't this generally referred to as the `patch` version?

Comment (luhenry) [reply internal-api/src/main/java/datadog/trace/api/Platform.java:67]: From the [JEP describing the version string](https://openjdk.java.net/jeps/322), this 3rd field is the "update" field. However, if we are being pedantic, `major` should then be "feature" and `minor` should be "interim". I'm happy to switch that to "patch" if you feel strongly about it.

Comment (jbachorik) [reply internal-api/src/main/java/datadog/trace/api/Platform.java:67]: Yeah, when you hear talking about JDK it is usually referred to as 'update' 🤷

**Review by jbachorik — APPROVED** [review]

File: internal-api/src/main/java/datadog/trace/api/Platform.java (hunk 1 above)
Comment (tylerbenson) [reply internal-api/src/main/java/datadog/trace/api/Platform.java:67]: ok, I'm sold then.

**Review by tylerbenson — APPROVED** [review]

**Pull request merged**